core/ecschema-metadata/src/SchemaJsonLocater.ts

```diff
@@ -41,7 +42,15 @@ export class SchemaJsonLocater implements ISchemaLocater {
    * @param matchType The match type to use when locating the schema
    */
   public async getSchemaInfo(schemaKey: Readonly<SchemaKey>, matchType: SchemaMatchType, context: SchemaContext): Promise<SchemaInfo | undefined> {
-    return this.getSchema(schemaKey, matchType, context);
+    const schemaProps = this._getSchema(schemaKey.name);
+    if (!schemaProps)
+      return undefined;
+
+    const schemaInfo = await Schema.startLoadingFromJson(schemaProps, context || new SchemaContext());
```

Comment:
`|| new SchemaContext()` seems unnecessary

Reply:
I think this was copied from the ECSchemaRpcLocater ... context used to be `SchemaContext | undefined`.  Since it should always be defined now I think it's fine to remove it

Reply:
I kept that in since getSchemaSync from SchemaJsonLocater also had this check.
I've removed it now.